algorithms/linfa-nn/src/distance.rs

```diff
@@ -77,6 +77,21 @@ impl<F: Float> Distance<F> for LInfDist {
     }
 }
 
+/// L0 or [Hamming](https://en.wikipedia.org/wiki/Hamming_distance)
+#[derive(Debug, Clone, PartialEq)]
+pub struct L0Dist;
```

Comment:
it kinda makes sense to continue 2/1/inf with zero norm, but like Yuhan said, it's not a real norm (and putting it into the framework of norms is misleading) - therefore distinguishing it from other norms by naming it after hamming code, makes it discrete behaviour more clear